File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s(nums.begin(), nums.end());
        int maxLength = 0;
        for (auto& num: nums) {
            if (s.find(num - 1) == s.end()) {
                // num is the start of a sequence
                int length = 1;
                int item = num+1;
                while (s.find(item) != s.end()) {
                    length++;
                    item++;
                }
                maxLength = max(maxLength, length);
            }
        }
        return maxLength;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-5.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        # sliding window??
        n = len(nums)
        if n == 1:
            return nums[0]
        rsum = 0
        res = nums[0]
        for num in nums:
            rsum += num
            res = max(res, rsum)
            if rsum < 0:
                rsum = 0
        return res 
        